src/app/view/components/login/dot-login-component/dot-login.component.ts

```diff
@@ -0,0 +1,181 @@
+import {
+    Component,
+    EventEmitter,
+    Input,
+    Output,
+    AfterViewInit,
+    OnInit,
+    ViewChild,
+    ElementRef
+} from '@angular/core';
+import { LoginService, LoggerService } from 'dotcms-js';
+import { DotLoginData, DotLoginLanguage, DotSystemInformation } from '@models/dot-login';
+import { SelectItem } from 'primeng/api';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+    selector: 'dot-login-component',
+    templateUrl: './dot-login.component.html',
+    styleUrls: ['./dot-login.component.scss']
+})
+/**
+ * The login component allows the user to fill all
+ * the info required to log in the dotCMS angular backend
+ */
+export class DotLoginComponent implements AfterViewInit, OnInit {
+    @Input() message = '';
+    @Input() passwordChanged = false;
+    @Input() resetEmailSent = false;
+    @Input() resetEmail = '';
+    @Input()
+    set isLoginInProgress(value: boolean) {
+        this.disableFormFields(value);
+    }
+    @Output() recoverPassword = new EventEmitter<any>();
+    @Output() login = new EventEmitter<DotLoginData>();
+
+    @ViewChild('emailInput') emailInput: ElementRef;
+
+    loginForm: FormGroup;
+    languages: SelectItem[] = [];
+    dotSystemInformation: DotSystemInformation;
+
+    dataI18n: { [key: string]: string } = {};
+
+    isCommunityLicense = true;
+    userIdOrEmailLabel = '';
+    emailAddressLabel = '';
+
+    private i18nMessages: Array<string> = [
+        'email-address',
+        'user-id',
+        'password',
+        'remember-me',
+        'sign-in',
+        'get-new-password',
+        'cancel',
+        'Server',
+        'error.form.mandatory',
+        'angular.login.component.community.licence.message',
+        'reset-password-success',
+        'a-new-password-has-been-sent-to-x',
+        'welcome-back'
+    ];
+
+    constructor(
+        private loginService: LoginService,
+        private loggerService: LoggerService,
+        private fb: FormBuilder
+    ) {}
+
+    ngOnInit() {
+        this.loginService.setAuth({
+            loginAsUser: null,
+            user: null,
+            isLoginAs: false
+        });
+        this.loginForm = this.fb.group({
+            login: ['', [Validators.required]],
+            language: [''],
+            password: ['', [Validators.required]],
+            rememberMe: false
+        });
+        this.renderPageData();
+    }
+
+    ngAfterViewInit(): void {
+        this.emailInput.nativeElement.focus();
+    }
+
+    /**
+     *  Executes the logIn service
+     */
+    logInUser(): void {
+        if (this.loginForm.valid) {
```

Comment:
the 'login' button is disabled when the form is not valid so really we need to check it here